scipy/datasets/meson.build

```diff
@@ -11,4 +11,6 @@ py3.install_sources(
   subdir: 'scipy/datasets'
 )
 
-subdir('tests')
+if not get_option('disable-network-tests')
+  subdir('tests')
+endif
```

Comment:
I'll let others chime in, but perhaps relevant to discussion is that I remember a previous conversation where, in a SciPy community meeting, we seemed less favorable toward the long-term concept of i.e., silently allowing the test suite to pass when the online/dataset assets were not available, because there was some concern it may conceal too many tests + some maintenance burden for both scenarios.

That said, this is a slightly different context since it requires explicit opt-in rather than, say a `pytest` `xfail` or `skip` directive approach. Formally, I still prefer the idea of using `pytest` mechanisms for test skipping/failure behavior because it then gets more appropriately reported in the final result tally, but I'm not so sure we'd want the extra effort of maintaining the marks for this particular opt-in scenario, so I'll be curious to see what others think.

Reply:
Is it possible to disable these tests via a pytest argument? If so, that would be sufficient.

Reply:
> Is it possible to disable these tests via a pytest argument? If so, that would be sufficient.

We'd have to mark all tests that try to download something, then we could skip those tests. I'd prefer this as well, skipping all tests sounds like a pretty big hammer (but then, I don't know what percentage of tests in `scipy/datasets/tests` would even work without an internet connection; if it's a high percentage, I personally think it would be OK)

Reply:
Most tests are and will be (in the future when we add/migrate more dataset methods) within [`TestDatasets`](https://github.com/scipy/scipy/blob/458cc70c8ba3a00daf3f3ad1d98695b95a894c45/scipy/datasets/tests/test_data.py#L27), which do require network access.

`test_clear_cache` is the only test that can still work without the internet.

I can send a PR by EOD today if everyone agrees to add a mark  (https://docs.pytest.org/en/7.1.x/example/markers.html#marking-test-functions-and-selecting-them-for-a-run) or use `pytest-remote` as discussed below (Note: Adds a test time dep) whichever is preferable.

I'm trying to recall if there was an argument against this; sounds fine to me as long as it is opt-in. Maybe @rgommers can share thoughts?